src/bun.js/api/bun/subprocess.zig

```diff
@@ -706,10 +707,11 @@ pub const Subprocess = struct {
 
     pub fn onStdinDestroyed(this: *Subprocess) void {
         this.flags.has_stdin_destructor_called = true;
-        this.weak_file_sink_stdin_ptr = null;
+        this.island.clearStdin();
 
         if (this.flags.finalized) {
             // if the process has already been garbage collected, we can free the memory now
```

Comment:
We don't need `state.finalized` anymore. This function should not be called when the subprocess has been collected.

Reply:
good catch